src/test/java/rx/internal/util/IndexedRingBufferTest.java

```diff
@@ -191,11 +191,7 @@ public Boolean call(String t1) {
             @Override
             public Boolean call(String t1) {
                 list.add(t1);
-                if (i++ == 2) {
-                    return false;
-                } else {
-                    return true;
-                }
+                return i++ != 2;
```

Comment:
>I could take that "i++" one separate line higher.

That's why I don't like such code, it's non-obvious (but short), in fact, "post `++`" will be applied after the condition check, so correct transformation would be:

```java
final boolean result = i != 2;
i++;
return result;
``` 

At least, we don't have undefined behavior here, so, you can leave it as is, but personally, I'd rewrite it.